mcs/class/System/Mono.AppleTls/Certificate.cs

```diff
@@ -331,7 +361,67 @@ public void Dispose ()
 
 		protected virtual void Dispose (bool disposing)
 		{
-			if (handle != IntPtr.Zero){
+			if (owner != IntPtr.Zero) {
+				CFObject.CFRelease (owner);
+				owner = handle = IntPtr.Zero;
+			} else if (handle != IntPtr.Zero) {
```

Comment:
It seems a tad weird to have this else in the Dispose. Not necessary wrong, but not expected.